test/Api.Test/SecretsManager/Controllers/RequestSMAccessControllerTests.cs

```diff
@@ -0,0 +1,82 @@
+
+[ControllerCustomize(typeof(RequestSMAccessController))]
+[SutProviderCustomize]
+public class RequestSMAccessControllerTests
+{
+    [Theory]
+    [BitAutoData]
+    public async Task RequestSMAccessFromAdminst_WhenSendingNoModel_ShouldThrowNotFoundException(
+    User user, SutProvider<RequestSMAccessController> sutProvider)
+    {
+        // Arrange
+        sutProvider.GetDependency<IUserService>().GetUserByPrincipalAsync(Arg.Any<ClaimsPrincipal>()).Returns(user);
+        sutProvider.GetDependency<IOrganizationRepository>().GetByIdentifierAsync(Arg.Any<string>()).ReturnsNullForAnyArgs();
+
+        // Act & Assert
+        await Assert.ThrowsAsync<NotFoundException>(() => sutProvider.Sut.RequestSMAccessFromAdmins(new RequestSMAccessRequestModel()));
+    }
+
+    [Theory]
+    [BitAutoData]
+    public async Task RequestSMAccessFromAdmins_WhenSendingValidData_ShouldSucceed(
+    User user,
+    RequestSMAccessRequestModel model,
+    Core.AdminConsole.Entities.Organization org,
+    ICollection<OrganizationUserUserDetails> orgUsers,
+    SutProvider<RequestSMAccessController> sutProvider)
+    {
+        // Arrange
+        sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(model.OrganizationId).Returns(org);
+        sutProvider.GetDependency<IUserService>().GetUserByPrincipalAsync(Arg.Any<ClaimsPrincipal>()).Returns(user);
+        sutProvider.GetDependency<IOrganizationUserRepository>().GetManyDetailsByOrganizationAsync(org.Id).Returns(orgUsers);
+        sutProvider.GetDependency<ICurrentContext>().OrganizationUser(model.OrganizationId).Returns(true);
+        // Act & Assert
+        await sutProvider.Sut.RequestSMAccessFromAdmins(model);
+
+        //Also check that the command was called
+        await sutProvider.GetDependency<IRequestSMAccessCommand>()
+            .Received(1)
+            .SendRequestAccessToSM(org, orgUsers, user, model.EmailContent);
+    }
+
+    [Theory]
+    [BitAutoData]
+    public async Task RequestSMAccessFromAdmins_WhenUserInvalid_ShouldThrowBadRequestException(RequestSMAccessRequestModel model, SutProvider<RequestSMAccessController> sutProvider)
+    {
+        // Arrange
+        sutProvider.GetDependency<IUserService>().GetUserByPrincipalAsync(Arg.Any<ClaimsPrincipal>()).ReturnsNullForAnyArgs();
+
+        // Act & Assert
+        await Assert.ThrowsAsync<UnauthorizedAccessException>(() => sutProvider.Sut.RequestSMAccessFromAdmins(model));
+    }
+
+    [Theory]
+    [BitAutoData]
+    public async Task RequestSMAccessFromAdmins_WhenOrgInvalid_ShouldThrowNotFoundException(RequestSMAccessRequestModel model, User user, SutProvider<RequestSMAccessController> sutProvider)
+    {
+        // Arrange
+        sutProvider.GetDependency<IOrganizationRepository>().GetByIdentifierAsync(Arg.Any<string>()).ReturnsNullForAnyArgs();
+        sutProvider.GetDependency<IUserService>().GetUserByPrincipalAsync(Arg.Any<ClaimsPrincipal>()).ReturnsForAnyArgs(user);
+
```

Comment:
```
 sutProvider.GetDependency<ICurrentContext>().OrganizationUser(model.OrganizationId).Returns(true);
```
Checking that the user is in the organization comes before checking if the Organization is valid. 

Need to add the mock to get to organization invalid call.